benchmarks/report_bayesmark.py

```diff
@@ -0,0 +1,92 @@
+from collections import Counter
+import io
+import json
+import os
+from typing import Dict
+
+
+_TABLE_HEADER = "\n|Ranking|Solver|Score|\n|---:|:---|---:|\n"
+
+
+class BayesmarkReportBuilder:
+    def __init__(self) -> None:
+
+        self._solvers: Counter = Counter()
+        self._datasets: Counter = Counter()
+        self._models: Counter = Counter()
+        self._problems_counter = 1
+        self._problems_body = io.StringIO()
+
+    def add_problem(self, name: str, scores: Dict[str, float]) -> "BayesmarkReportBuilder":
+
+        if self._problems_body.closed:
+            self._problems_body = io.StringIO()
+
+        problem_title = f"### ({self._problems_counter}) Problem: {name}\n"
+        self._problems_body.write(problem_title)
+        self._problems_body.write(_TABLE_HEADER)
+
+        for idx, (solver, score) in enumerate(scores.items()):
+            row = f"|{idx + 1}|{solver}|{score:.5f}|\n"
+            self._problems_body.write(row)
+
+        self._solvers.update(scores.keys())
+        self._problems_counter += 1
+
+        return self
+
+    def add_dataset(self, dataset: str) -> "BayesmarkReportBuilder":
+
+        self._datasets.update([dataset])
+        return self
+
+    def add_model(self, model: str) -> "BayesmarkReportBuilder":
+
+        self._models.update([model])
+        return self
+
+    def assemble_report(self) -> str:
+
+        num_datasets = len(self._datasets)
+        num_models = len(self._models)
+
+        with open(os.path.join("benchmarks", "bayesmark", "report_template.md")) as file:
+            report_template = file.read()
+
+        # TODO(xadrianzetx) Consider using proper templating engine.
```

Comment:
:+1: for it. We may be able to use a template engine such as jinja. What concerns does it block the task (I think the current implementation is fine, just for curiosity)?

Reply:
Yeah, I think we are good for now. Template engine would be nice if we wanted to add more fidelity to the report (like e.g. embedding plots or using different font colors) later on.